src/main/java/org/sonarlint/intellij/ui/tree/SecurityHotspotTreeModelBuilder.java

```diff
@@ -234,11 +235,20 @@ public int applyCurrentFiltering() {
     return filterSecurityHotspots(currentFilter);
   }
 
+  public int updateStatusAndApplyCurrentFiltering(String securityHotspotKey, HotspotStatus status) {
+    for (var securityHotspotNode : nonFilteredNodes) {
+      if (securityHotspotKey.equals(securityHotspotNode.getHotspot().getServerFindingKey())) {
+        securityHotspotNode.getHotspot().setStatus(status);
```

Comment:
I think we can probably add a `break` here, security hotspots should be unique so no need to keep going if we found one